src/libasr/codegen/llvm_utils.cpp

```diff
@@ -2365,8 +2365,18 @@ namespace LCompilers {
         // TODO: Should be created outside the user loop and not here.
         // LLVMList should treat them as data members and create them
         // only if they are NULL
+        llvm::Value* adjusted_pos = builder->CreateSelect(
+            builder->CreateICmpSGT(pos, current_end_point),
+            current_end_point,
+            pos
+        );
+        adjusted_pos = builder->CreateSelect(
+            builder->CreateICmpSGE(adjusted_pos, llvm::ConstantInt::get(context, llvm::APInt(32, 0))),
+            adjusted_pos,
+            llvm::ConstantInt::get(context, llvm::APInt(32, 0))
+        );
         llvm::AllocaInst *tmp_ptr = builder->CreateAlloca(el_type, nullptr);
```

Comment:
The TODO at line 2365 is for AllocaInst, I think. Why don't you move this above?

Reply:
Okay sure. You are right.